question-generation/backend.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+from transformers import BartTokenizer, BartForConditionalGeneration
+from pathlib import Path
+import random
+import torch
+
+
+def load_contexts_from(file_name: Path) -> "list[str]":
+    """Retrieve context sequences from a text file.
+
+    Args:
+        file_name: A path to the text file containing the context sequences.
+          Each line represents a separate sequence.
+
+    Returns:
+        A list of context sequences as strings.
+
+    Raises:
+        FileNotFoundError: file_name does not exist.
+    """
+
+    if file_name.is_file():
+        with open(file_name, "r") as f:
+            return f.readlines()
+    else:
+        raise FileNotFoundError(f"'{file_name.name}' does not exist.")
+
+
+def generate_questions(sequence: str) -> "list[str]":
+    """Generate questions based on a given sequence of context.
+
+    Args:
+        sequence: A string representing the context from which to generate questions.
+
+    Returns:
+        A list of generated questions as strings. The number of questions returned
+        is determined by the num_return_sequences parameter of the model's generate method.
+    """
+
+    model_id = "hyechanjun/interview-question-remake"
+    tokenizer = BartTokenizer.from_pretrained(model_id)
+    model = BartForConditionalGeneration.from_pretrained(model_id)
```

Comment:
Loading a model can be expensive in time and GPU memory. The API of this function does not permit using an already-loaded model. You could consider using the Huggingface Pipeline API to handle all of this.

Reply:
Huggingface Pipeline will be implemented.